students/bcamp3/lesson04/trigrams.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python3
+"""Trigrams assignment."""
+import sys
+import random
+import string
+
+# test out words
+# test_words = ("I wish I may I wish I might I wish I could go home tonight"
+#               " I wish upon a star tonight visible from home a star upon my"
+#               " sight".split()
+#               )
+
+
+def read_in_data(filename):
+    """Read text file."""
+    body = False
+    data = ''
+    with open(filename, 'r') as file:
+        for line in file:
+            if line[:9] == '*** START':
+                body = True
+            if line[:7] == '*** END':
+                body = False
+            if body and not line.isupper():
```

Comment:
At this line the 'body' of the book needs to be identified for data to be read-in.  If 'body' is not found, data is not read.